rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy

```diff
@@ -71,30 +71,26 @@ abstract class CloudProviderBakeHandler {
    * bake.
    */
   String produceBakeKey(String region, BakeRequest bakeRequest) {
-    // TODO(duftler): Consider whether template_file_name, extended_attributes and var_file_name should be
-    // taken into account when composing the bake key.
     bakeRequest.with {
-      def bakeKey = "bake:$cloud_provider_type:$base_os"
-
-      if (base_ami) {
-        bakeKey += ":$base_ami"
-      }
-
-      if (ami_name) {
-        bakeKey += ":$ami_name"
-      }
-
-      String packages = package_name ? package_name.tokenize().join('|') : ""
-
-      bakeKey += ":$packages"
-
-      def providerSpecificBakeKeyComponent = produceProviderSpecificBakeKeyComponent(region, bakeRequest)
-
+      def keys = [
+        'bake',
+        cloud_provider_type,
+        base_os,
+        base_ami,
+        ami_name,
+        template_file_name,
+        var_file_name,
+      ].findAll { it }
+
+      // Package name is always part of key, even if it is an empty string
+      keys << (package_name ? package_name.tokenize().join('|') : "")
```

Comment:
Shouldn't this be access from `bakeRequest`. If not, this variable is empty and the package will be always `""`, right?